### DISCUSSION TITLE
Data type for generic card?

### DISCUSSION THREAD

**Discussion by mtanco** [h2oai/wave#918] [Q&A] [answered]
Is there an appropriate way to say that the datatype of the `card` parameter will be a `ui.*_card`?

```python
def add_card_to_page(q: Q, card, card_name: str) -> None:
    """
    Display a card which we would want to remove on page change
    """
    q.page[card_name] = card
    q.client.active_cards.append(card_name)
```

**Comment by lo5** [comment, marked as answer]
The cards don't inherit from a common base class. Anything that is either a `dict` or an object with a `dump() -> dict` method can be added to a page. The API uses `Any` since I'm not aware of a precise way to express this using `typing`.
